### PR TITLE
Restore beat

### PR DESCRIPTION
**Pull request by jeromegrosse**
## Status
**READY**

## Migrations
NO

## Description
Restore beat command

branch | PR
------ | ------
restore_beat | [link](https://github.com/ticketea/odin/pull/94)
sqjobs_odin | [link](https://github.com/ticketea/devops/pull/203)

### CODE REVIEW

File: sqjobs/contrib/django/djsqjobs/management/commands/sqjobs.py [hunk 1]
```diff
@@ -10,51 +12,54 @@ class Command(BaseCommand):
...
 
     VALID_MODE_WORKER = "worker"
+    VALID_MODE_BEAT = "beat"
     VALID_TRANSPORT_SQS = "sqs"
 
     def add_arguments(self, parser):
-        parser.add_argument("transport-or-mode", type=str, choices=[self.VALID_TRANSPORT_SQS, self.VALID_MODE_WORKER])
-        parser.add_argument("mode-or-queue-name", type=str)
-        parser.add_argument("queue-name", type=str, default=None, nargs="*")
+        parser.add_argument("transport", type=str, choices=[self.VALID_TRANSPORT_SQS])
```
Comment (Kartones) [inline sqjobs/contrib/django/djsqjobs/management/commands/sqjobs.py]: test very carefully before deploying live this change, in theory we already are calling with the new format, but all this ifs and checks had to be added both due to sqjobs version changes and django issues

**Comment by Kartones** [comment]
Can you give some background on what is this `Beat`? why a cron-like custom implementation of job runner?

File: sqjobs/contrib/django/djsqjobs/management/commands/sqjobs.py (hunk 1 above)
Comment (jeromegrosse) [reply sqjobs/contrib/django/djsqjobs/management/commands/sqjobs.py]: Sure, there is an upcoming PR in `devops` that address this very issue. We are currently NOT calling it with the new form but with the legacy one.
I'll add you to the PR of `devops` so you can check it out.

**Comment by jeromegrosse** [comment]
Sure, so the way I understand it, beats are basically like CRONs for SQJobs. You'll execute each X seconds a command. In our case, it is a command that will triggers the generation of a report and sending it to a mail address. I'm sure @gnufede would have some insight to give on the matter.

**Comment by jeromegrosse** [comment]
The tests are finally working.

Regarding the usage of CRONs: yeah, I hear you totally. The things is as you said, the code was already there and I didn't want to reinvent the wheel. I've been pulling the thread a lot up to the point to refactor a part of SQJobs and I think that this is as far as the scope of this task goes.

Hopefully, soon Odin will be shutdown and we'll be able to go with a more standard way of performing reports.

**Pull request merged**